File: backend/src/models/Event.ts
import mongoose, { Schema } from 'mongoose';

export type EventCategory = 'workshop' | 'seminar' | 'cultural' | 'sports' | 'technical' | 'other';

export interface IEvent extends mongoose.Document {
  title: string;
  description: string;
  category: EventCategory;
  date: Date;
  endDate?: Date;
  venue: string;
  capacity: number;
  registeredCount: number;
  banner?: string;
  organizer: mongoose.Types.ObjectId;
  tags: string[];
  isActive: boolean;
  isCancelled: boolean;
  cancelReason?: string;
  registrationDeadline?: Date;
  createdAt: Date;
  updatedAt: Date;
}

const eventSchema = new Schema<IEvent>(
  {
    title: { type: String, required: true, trim: true },
    description: { type: String, required: true },
    category: {
      type: String,
      required: true,
      enum: ['workshop', 'seminar', 'cultural', 'sports', 'technical', 'other']
    },
    date: { type: Date, required: true },
    endDate: { type: Date },
    venue: { type: String, required: true },
    capacity: { type: Number, required: true, min: 1 },
    registeredCount: { type: Number, default: 0, min: 0 },
    banner: { type: String },
    organizer: { type: Schema.Types.ObjectId, ref: 'User', required: true },
    tags: [{ type: String }],
    isActive: { type: Boolean, default: true },
    isCancelled: { type: Boolean, default: false },
    cancelReason: { type: String },
    registrationDeadline: { type: Date }
  },
  { timestamps: true }
);

eventSchema.index({ title: 'text', description: 'text', tags: 'text' });

const Event = mongoose.model<IEvent>('Event', eventSchema);

export default Event;

File: backend-spring/src/main/java/com/campusconnect/model/Event.java
package com.campusconnect.model;

import lombok.Data;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.index.TextIndexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;

@Data
@Document(collection = "events")
public class Event {

    @Id
    private String id;

    @TextIndexed
    private String title;

    @TextIndexed
    private String description;

    private String category; // workshop, seminar, cultural, sports, technical, other

    private Instant date;

    private Instant endDate;

    private String venue;

    private int capacity;

    private int registeredCount;

    private String banner;

    private String organizer; // User id

    @TextIndexed
    private List<String> tags = new ArrayList<>();

    private boolean isActive = true;

    private boolean isCancelled;

    private String cancelReason;

    private Instant registrationDeadline;

    @CreatedDate
    private Instant createdAt;

    @LastModifiedDate
    private Instant updatedAt;
}
